### DISCUSSION TITLE
Bind port on edge anchor and apply it on every nodes/edges change.

### DISCUSSION THREAD

**Discussion by niemes** [antvis/X6#1851] [Q&A] [answered]
Hi,

I'm searching a way to bind the position of a port on a point of the edge targeting a node.
As an example : 

I have : 
- 1 circle node
- 1 edge , targetting a node using an anchor: nodeCenter.
- 1 port

![FollowPort_img](https://i.imgur.com/YnPiInV.gif)

So far I ended up calculating the coordinates of the anchor and applied them to the port, but I'm not sure about the best option to do this.
 
```javascript
this.node.prop('ports/groups/group1/position', {
        name: "absolute",
        args: {
          x: this.graph.findViewByCell(edge1.id).targetPoint.x - this.node.position().x,
          y: this.graph.findViewByCell(edge1.id).targetPoint.y - this.node.position().y
        },
      })
```

I would like to apply this method every time the nodes/edges move. Also, this method seems quite slow and doesn't consider the circle to correctly map the port on the edge... Any ideas ?

Thanks for your help and this awesome library!

**Comment by NewByVector** [comment, marked as answer]
I wrote a [demo](https://codesandbox.io/s/rotate-port-www60p?file=/src/app.tsx) with a different idea,take a look if you're interested.

**Reply by niemes** [reply]
That's a very elegant solution. I like it. I struggled to find the docs about the Vectors. 

I ended up changing the marker after the connection while removing the port and using anchor 😅. I prefer your solution way cleaner.

Thank you for taking the time @NewByVector . 👍😊

**Reply by NewByVector** [reply]
Thank you for your reply.
